spec/controllers/orgs/rosters_controller_spec.rb

```diff
@@ -261,6 +242,189 @@
     end
   end
 
+  describe "GET #select_google_classroom", :vcr do
+    before do
+      sign_in_as(user)
+      GoogleAPI = Google::Apis::ClassroomV1
+    end
+
+    context "with student identifier flipper enabled" do
+      before do
+        GitHubClassroom.flipper[:student_identifier].enable
+      end
+
+      context "with google classroom flipper enabled" do
+        before do
+          GitHubClassroom.flipper[:google_classroom_roster_import].enable
+        end
+
+        context "when user is authorized with google" do
+          before do
+            # Stub google authentication again
+            client = Signet::OAuth2::Client.new
+            allow_any_instance_of(Orgs::RostersController)
+              .to receive(:user_google_classroom_credentials)
+              .and_return(client)
+
+            # Stub list courses response
+            response = GoogleAPI::ListCoursesResponse.new
+            allow_any_instance_of(GoogleAPI::ClassroomService)
+              .to receive(:list_courses)
+              .and_return(response)
+
+            get :select_google_classroom, params: {
+              id: organization.slug
+            }
+          end
+
+          it "succeeds" do
+            expect(response).to have_http_status(:success)
+          end
+        end
+
+        context "when user is not authorized with google" do
+          before do
+            allow_any_instance_of(Orgs::RostersController)
+              .to receive(:user_google_classroom_credentials)
+              .and_return(nil)
+
+            get :select_google_classroom, params: {
+              id: organization.slug
+            }
+          end
+
+          it "redirects to authorization url" do
+            expect(response).to redirect_to %r{\Ahttps://accounts.google.com/o/oauth2}
+          end
+        end
+
+        after do
+          GitHubClassroom.flipper[:google_classroom_roster_import].disable
+        end
+      end
+
+      context "with google classroom identifier disabled" do
+        before do
+          get :search_google_classroom, params: {
+            id: organization.slug,
+            query: ""
+          }
+        end
+
+        it "404s" do
+          expect(response).to have_http_status(:not_found)
+        end
+      end
+
+      after do
+        GitHubClassroom.flipper[:student_identifier].disable
+      end
+    end
+
+    context "with flipper disabled" do
+      before do
+        get :show, params: { id: organization.slug }
+      end
+
+      it "404s" do
+        expect(response).to have_http_status(:not_found)
```

Comment:
Why is `:show` 404ing here? Shouldn't `show` render fine regardless of whether or not the flipper is enabled?

Reply:
Yeah we should be linking to the Google Classroom setup instead. Not sure why this is passing. Just tested it manually and it isn't 404ing 😕

Reply:
Could you dig into why that's happening in the test? That's suspicious...

Reply:
This test was passing because we did not enable the `:student_identifier` flag. We should really get rid of all the feature flags that been at 100% for a while.

Reply:
We also already have a test covering the google class setup route. So I'll just modify this test to pass if the `:google_classroom_roster_import` is disabled.